napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,510 @@
+        To reduce the graphical artifacts, the inner vectors are limited
+        to half of the length of the adjacent egdes in path.
+    """
+    normals = np.empty_like(path)
+    vec_len_arr = np.empty((len(path)), dtype=np.float32)
+    for i in range(1, len(path)):
+        vec_diff = path[i] - path[i - 1]
+        vec_len_arr[i - 1] = np.sqrt(vec_diff[0] ** 2 + vec_diff[1] ** 2)
+        normals[i - 1] = vec_diff / vec_len_arr[i - 1]
+
+    if closed:
+        vec_diff = path[0] - path[-1]
+        vec_len_arr[-1] = np.sqrt(vec_diff[0] ** 2 + vec_diff[1] ** 2)
+        normals[-1] = vec_diff / vec_len_arr[-1]
+
+    return normals, vec_len_arr * 0.5
+
+
+@njit(cache=True, inline='always')
+def _generate_2D_edge_meshes_loop(
+    path: np.ndarray,
+    closed: bool,
+    cos_limit: float,
+    bevel: bool,
+    normals: np.ndarray,
+    bevel_limit_array: np.ndarray,
+    centers: np.ndarray,
+    offsets: np.ndarray,
+    triangles: np.ndarray,
+) -> None:
+    """Main loop for :py:func:`generate_2D_edge_meshes`.
+
+    Parameters
+    ----------
+    path : np.ndarray
+        Nx2 array of central coordinates of path to be triangulated
+    closed : bool
+        Bool which determines if the path is closed or not
+    cos_limit : float
+        Miter limit which determines when to switch from a miter join to a
+        bevel join
+    bevel : bool
+        Bool which if True causes a bevel join to always be used. If False
+        a bevel join will only be used when the miter limit is exceeded
+    normals : np.ndarray
+        Nx2 array of normal vectors of the path
+    bevel_limit_array : np.ndarray
+        The array of limit of length of inner vectors in bevel joins.
+        To reduce the graphical artifacts, the inner vectors are limited
+        to half of the length of the adjacent edges in path.
+    centers : np.ndarray
+        Mx2 array to put central coordinates of path triangles.
+    offsets : np.ndarray
+        Mx2 array to put the offsets to the central coordinates that need to
+        be scaled by the line width and then added to the centers to
+        generate the actual vertices of the triangulation
+    triangles : np.ndarray
+        (M-2)x3 array to put the indices of the vertices that will form the
+        triangles of the triangulation
+    """
+    if closed:
+        j = _set_centers_and_offsets(
+            centers,
+            offsets,
+            triangles,
+            path[0],
+            normals[-1],
+            normals[0],
+            bevel_limit_array[-1],
+            bevel_limit_array[0],
+            0,
+            cos_limit,
+            bevel,
+        )
+    else:
+        centers[0] = path[0]
+        centers[1] = path[0]
+        offsets[0, 0] = normals[0][1] * 0.5
+        offsets[0, 1] = -normals[0][0] * 0.5
+        offsets[1] = -offsets[0]
+        triangles[0] = [0, 1, 2]
+        triangles[1] = [1, 2, 3]
+        j = 2
+    for i in range(1, len(normals) - 1):
```

Comment:
```suggestion

    for i in range(1, len(normals) - 1):
```

Add new line to separate the else block.